src/gt4py/next/program_processors/runners/dace_iterator/workflow.py

```diff
@@ -204,8 +206,12 @@ def convert_args(
     on_gpu = True if device == core_defs.DeviceType.CUDA else False
 
     def decorated_program(
```

Comment:
This function is where the PR could have an impact. Question is if - with this PR - `decorated_program` will still be called every time the program is run, even when the program is AOT-compiled and cached?

Reply:
Yes, it is the only way to provide the same interface. Without this, the user would have to
- not pass "out" as a kwarg to programs generated from field operators
- pass implicit domain size arguments manually

Reply:
Functionality could be added to make it possible to circumvent if the user can guarantee the above.